### PR TITLE
Add support for impersonation chain to GKEStartPodOperator in deferrable mode

### PR DESCRIPTION
**Pull request by m1racoli**
We utilize the existing implementation of `_CredentialsToken` by using the async hook's `get_token` method. This implementation allows us to leverage several features of the Google connection from `Keyfile Path` or `Keyfile JSON` (see #37081) to impersonation chain on hook or connection level. We therefore do not need to rely on the async hook's `service_file_as_context` method, which does not support impersonation chain.

With this change we effectively gain support for impersonation chain in GKEStartPodOperator in deferrable mode.

related: #37081 

Adding @Lee-W for visibility.

### CODE REVIEW

File: airflow/providers/google/cloud/hooks/kubernetes_engine.py [hunk 1]
```diff
@@ -709,37 +709,34 @@ async def get_pod(self, name: str, namespace: str) -> V1Pod:
...
-        with await self.service_file_as_context() as service_file:  # type: ignore[attr-defined]
-            async with Token(scopes=self.scopes, service_file=service_file) as token:
-                async with self.get_conn(token) as connection:
-                    v1_api = async_client.CoreV1Api(connection)
-                    pod: V1Pod = await v1_api.read_namespaced_pod(
-                        name=name,
-                        namespace=namespace,
-                    )
-                return pod
+        token = await self.get_token()
```
Comment (Lee-W) [inline airflow/providers/google/cloud/hooks/kubernetes_engine.py:712]: I kinda forget the detail. Will we still be able to authorize through key file and key json in this case?

Comment (m1racoli) [reply airflow/providers/google/cloud/hooks/kubernetes_engine.py:712]: Yes, we use [hook.get_credentials_and_project_id()](https://github.com/apache/airflow/blob/fa234e0d229dfebb6284dd851870e64186fb53e2/airflow/providers/google/common/hooks/base_google.py#L661C37-L661C67) to generate credentials used in `_CredentialsToken`. This method respects what is configured in the GCP connection and thus also supports keyfile path and keyfile dict or `GOOGLE_APPLICATION_CREDENTIALS`.

Comment (m1racoli) [reply airflow/providers/google/cloud/hooks/kubernetes_engine.py:712]: In general we should replace all uses of `gcloud.aio.auth.token.Token` with our `_CredentialsToken` implementation across the provider (there are only 2 or 3 uses left) to provide consistent support of async operations with credentials from GCP connections.

Later on, once we have upgraded `gcloud-aio` to the lastest major version we can look into replacing our `_CredentialsToken` with gcloud-aio's Token by refactoring [GoogleBaseAsyncHook.get_token](https://github.com/apache/airflow/blob/fa234e0d229dfebb6284dd851870e64186fb53e2/airflow/providers/google/common/hooks/base_google.py#L696) appropriately.

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
LGTM: @Lee-W ?

**Review by Lee-W — APPROVED** [review]
Sorry for late reply. LGTM. 👍

**Comment by thomasLeclaire** [comment]
Hello! Is there any reason to not also add such mechanism for the equivalent GKEStartJobOperator?

**Pull request merged**